src/types.rs

```diff
@@ -238,33 +238,55 @@ pub struct NetworkIdentity {
     pub metadata: MetaData,
 }
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Debug)]
 pub enum PeerState {
     Disconnected,
     Connecting,
     Connected,
     Disconnecting,
 }
 
-#[derive(Serialize, Deserialize)]
+impl fmt::Display for PeerState {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let printable = match *self {
+            Self::Disconnected => "disconnected",
+            Self::Connecting => "connecting",
+            Self::Connected => "connected",
+            Self::Disconnecting => "disconnecting",
+        };
+        write!(f, "{}", printable)
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
 pub enum ConnectionOrientation {
     Inbound,
     Outbound,
 }
 
+impl fmt::Display for ConnectionOrientation {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let printable = match *self {
+            Self::Inbound => "inbound",
+            Self::Outbound => "outbound",
+        };
+        write!(f, "{}", printable)
+    }
+}
+
 #[derive(Serialize, Deserialize)]
 pub struct PeerDescriptor {
     pub state: PeerState,
     pub direction: ConnectionOrientation,
 }
 
-#[derive(Serialize, Deserialize)]
+#[derive(Serialize, Deserialize, Debug)]
 pub struct PeerDescription {
     pub peer_id: PeerId,
     pub enr: Enr,
     pub last_seen_p2p_address: Multiaddr,
-    pub state: PeerState,
-    pub direction: ConnectionOrientation,
+    pub state: String,
+    pub direction: String,
```

Comment:
I think we can leave these as is to get the stronger type safety

I'm guessing there was a serialization issue as the names didn't match the enum variants?

if so there is a way to add a serde attribute that "renames" everything to lowercase, see here for more details: https://serde.rs/container-attrs.html#rename_all

Reply:
aah of course! I went round in circles on this - thanks. Now fixed.